streams/core.py

```diff
@@ -59,6 +59,13 @@ def __init__(self, child=None, children=None, **kwargs):
             if child:
                 child.parents.append(self)
 
+    def __str__(self):
+        s = self.__class__.__name__
+        for m in ['func', 'predicate']:
+            if hasattr(self, m):
+                s += ', {}'.format(getattr(self, m).__name__)
+        return s
```

Comment:
It might be nice to bound this with `<...>`, which might help disambiguate things if, for example, there are many of them in a list.  Similarly I suggest using colon instead of a comma.